api/src/main/java/com/basistech/rosette/api/RosetteAPI.java

```diff
@@ -283,7 +285,10 @@ private void initHttpClient() {
         if (customHeaders.size() > 0) {
             defaultHeaders.addAll(customHeaders);
         }
-        httpClient = HttpClients.custom().setDefaultHeaders(defaultHeaders).build();
+
+        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager();
+        cm.setMaxTotal(maxSockets);
+        httpClient = HttpClients.custom().setConnectionManager(cm).setDefaultHeaders(defaultHeaders).build();
```

Comment:
This code gets run once at initialization, and updating maxSockets doesn't update the connection manager.